elastic/beats
# [Stack Monitoring] Put `elasticsearch.node.stats.jvm.mem.pools.*`

This re-adds the `pools` to the `node_stats` metric set.

## Proposed commit message

Adds the `elasticsearch.node.stats.jvm.mem.pools.*` fields to the `metricset` for `node_stats` because it was removed by accident in a previous release.

## Checklist


## Disruptive User Impact

No. The mappings have continued to exist.

## Author's Checklist

- [ ] Verify that the data appears in the `node_stats` metricset

## How to test this PR locally

Configure Metricbeat:

```yaml
metricbeat.modules:
- module: elasticsearch
  metricsets:
    - node_stats
  period: 10s
  hosts: ["http://es:9200"]

output.elasticsearch:
  hosts: ["http://es:9200"]
```

Connect to Elasticsearch and observe the data stream provides

```http
GET /.ds-metricbeat-*/_search?q=metricset.name:node_stats&filter_path=hits.hits._source.elasticsearch.node.stats.jvm.mem.pools.*
```

Without this change, this does not exist in the resulting `_source`. With this change, you should see something along the lines of

```json
{
  "hits" : {
    "hits" : [
      {
        "_source" : {
          "elasticsearch" : {
            "node" : {
              "stats" : {
                "jvm" : {
                  "mem" : {
                    "pools" : {
                      "survivor" : {
                        "peak" : {
                          "bytes" : 74424848
                        },
                        "peak_max" : {
                          "bytes" : 0
                        },
                        "used" : {
                          "bytes" : 12582912
                        },
                        "max" : {
                          "bytes" : 0
                        }
                      },
                      "old" : {
                        "used" : {
                          "bytes" : 90397592
                        },
                        "max" : {
                          "bytes" : 1073741824
                        },
                        "peak" : {
                          "bytes" : 91899520
                        },
                        "peak_max" : {
                          "bytes" : 1073741824
                        }
                      },
                      "young" : {
                        "peak" : {
                          "bytes" : 624951296
                        },
                        "peak_max" : {
                          "bytes" : 0
                        },
                        "used" : {
                          "bytes" : 293601280
                        },
                        "max" : {
                          "bytes" : 0
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    ]
  }
}
```

## Related issues

Closes #40539

## Use cases

Understand the different memory pools in the JVM.
